src/main.cpp

```diff
@@ -3548,12 +3548,22 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
     int64_t nTime1 = GetTimeMicros(); nTimeConnect += nTime1 - nTimeStart;
     LogPrint("bench", "      - Connect %u transactions: %.2fms (%.3fms/tx, %.3fms/txin) [%.2fs]\n", (unsigned)block.vtx.size(), 0.001 * (nTime1 - nTimeStart), 0.001 * (nTime1 - nTimeStart) / block.vtx.size(), nInputs <= 1 ? 0 : 0.001 * (nTime1 - nTimeStart) / (nInputs-1), nTimeConnect * 0.000001);
 
-    CAmount blockReward = nFees + consensusParams.GetBlockSubsidy(pindex->nHeight);
-    if (block.vtx[0].GetValueOut() > blockReward)
+    CAmount cbTotalOutputValue = block.vtx[0].GetValueOut() + pindex->nLockboxValue;
```

Comment:
Good catch, this was indeed missing from the lockbox PR.

Or was it? Did `GetBlockSubsidy` have `nLockboxValue` pre-subtracted from it internally?

Reply:
`GetBlockSubsidy` does not take into account (i.e. does not subtract) `nLockboxValue`. That's correct because "block subsidy" as defined in the spec doesn't either.